### PR TITLE
Add Dawn Wallet

### PR DESCRIPTION
**Pull request by thomas-waite**
# Summary
Integrates Dawn Wallet into ConnectKit. Dawn Wallet is an iOS based Ethereum Wallet, with a native Safari extension: https://twitter.com/dawn_wallet 

It is an injected wallet and injects a provider into the web page, through which dapps then prompt connection requests and transaction signing through a Safari popup.

I've tested that this works and you can successfully connect using Dawn across the examples given: `nextjs`, `create react app` and `vite`.

Connect UI looks like:
image

### CODE REVIEW

File: packages/connectkit/src/wallets/connectors/dawn.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      default: <Logos.Dawn />,
+      mobile: <Logos.Dawn background />,
+      transparent: <Logos.Dawn background={false} />,
+      appIcon: <Logos.Dawn background={false} />,
+      connectorButton: <Logos.Dawn background={true} />,
+      qrCode: <Logos.Dawn background={true} />,
+    },
+    logoBackground: 'var(--ck-brand-dawn)',
+    scannable: true,
+    installed: undefined,
```
Comment (thomas-waite) [inline packages/connectkit/src/wallets/connectors/dawn.tsx:26]: To discuss `installed: undefined`

**Comment by thomas-waite** [comment]
Dependent on https://github.com/family/connectkit/pull/161 being merged. Needs the `wagmi` package version `0.12.0` - it has the `isDawn` flags in

**Comment by thomas-waite** [comment]
> Dependent on #161 being merged. Needs the `wagmi` package version `0.12.0` - it has the `isDawn` flags in

This is now done, PR ready for review. Have tested it works. The connect UI looks like:

image

File: examples/cra/package.json [hunk 2]
```diff
@@ -16,7 +16,7 @@
     "react-dom": "^18.0.0",
     "react-scripts": "5.0.1",
     "typescript": "^4.9.5",
-    "wagmi": "^0.12.0",
+    "wagmi": "^0.12.5",
```
Comment (thomas-waite) [inline examples/cra/package.json:19]: The wagmi package I needed was 0.12.4 I think (it has the `isDawn` flags) and `^0.12.0` for some reason wasn't installing the latest package version. So manually forced it to have the next minimum version I needed

File: examples/nextjs/package.json [hunk 3]
```diff
@@ -14,7 +14,7 @@
     "next": "12.3.0",
     "react": "^18.0.0",
     "react-dom": "^18.0.0",
-    "wagmi": "^0.12.0"
+    "wagmi": "^0.12.5"
```
Comment (thomas-waite) [inline examples/nextjs/package.json:17]: Same reason here and throughout for the `package.json`s

**Review by lochie — APPROVED** [review]
lgtm

**Comment by lochie** [comment]
@thomas-waite sorry for the delay on this PR. this will be available in connectkit `1.2.4`, we really appreciate the contribution! 😊

**Pull request merged**